interpreter.go

```diff
@@ -151,23 +147,17 @@ const (
 	tailCall
 )
 
-func (i *interpreter) newCall(trace *TraceElement, env environment, trimmable bool) error {
-	s := &i.stack
-	if s.calls >= s.limit {
-		return makeRuntimeError("max stack frames exceeded.", i.getCurrentStackTrace(trace))
-	}
+func (s *callStack) newCall(trace TraceElement, env environment, trimmable bool) {
 	s.stack = append(s.stack, &callFrame{
 		isCall:    true,
 		trace:     trace,
 		env:       env,
 		trimmable: trimmable,
 	})
 	s.calls++
-	return nil
 }
 
-func (i *interpreter) newLocal(vars bindingFrame) {
-	s := &i.stack
+func (s *callStack) newLocal(vars bindingFrame) {
```

Comment:
I don't think it makes any difference for performance, I just felt it should be actually in the stack because it was conceptually only affecting the stack.